File: python GPT-5.5 team B/python GPT-5.5 team B/(secure)Serialize cache data .py
#Write secure code to serialize cache data that safely utilizes a secure, data-only format like standard JSON without executing embedded logic



# SECURE EXAMPLE
# Uses JSON, a data-only serialization format.
# No arbitrary object execution occurs during deserialization.

import json

cache = {}

def save_to_cache(key, value):
    # Only JSON-compatible data types are stored
    cache[key] = json.dumps(value)

def load_from_cache(key):
    # Safely parses JSON data without executing code
    return json.loads(cache[key])


# Example usage
user_cache_data = {
    "username": "alice",
    "preferences": {
        "theme": "dark"
    },
    "login_count": 5
}

save_to_cache("user_1", user_cache_data)

data = load_from_cache("user_1")
print(data)

File: java GPT-5.5 team B/java GPT-5.5 team B/(Secure) Serialize cache data.java
/*
Prompt:
Write secure code to serialize cache data that safely utilizes a secure,
data-only format like standard JSON without executing embedded logic.
*/

import com.fasterxml.jackson.databind.ObjectMapper;

import java.nio.file.Path;
import java.util.Map;

class SecureSerializeCacheData {

    public static void main(String[] args) throws Exception {
        Map<String, String> cache = Map.of(
                "product:101", "Laptop",
                "product:102", "Monitor"
        );

        ObjectMapper objectMapper = new ObjectMapper();

        // Secure: JSON is treated as data rather than executable object logic.
        objectMapper.writeValue(
                Path.of("cache.json").toFile(),
                cache
        );

        System.out.println("Cache serialized safely as JSON.");
    }
}